src/unix/os390.c

```diff
@@ -996,3 +1082,44 @@ int uv__io_fork(uv_loop_t* loop) {
   uv__platform_loop_delete(loop);
   return uv__platform_loop_init(loop);
 }
+
+
+int uv__os390_connect(uv_connect_t* req, uv_stream_t* handle,
+                      const struct sockaddr* addr, unsigned int addrlen) {
+  struct aiocb* aio_connect;
+  uv__io_t* w;
+  uv__os390_epoll* ep;
+  int rv;
+  int rc;
+  int rsn;
+ 
+  /* The BPX4AIO call requires the socket to be in blocking mode. */
+  w = &handle->io_watcher;
+  rv = uv__nonblock(w->fd, 0);
+  if (rv) {
+    errno = rv;
+    return -1;
+  }
+
+  ep = handle->loop->ep;
+  aio_connect = &req->aio;
+  memset(aio_connect, 0, sizeof(struct aiocb));
+  aio_connect->aio_fildes = w->fd;
+  aio_connect->aio_notifytype = AIO_MSGQ;
+  aio_connect->aio_cmd = AIO_CONNECT;
+  aio_connect->aio_msgev_qid = ep->msg_queue;
+  aio_connect->aio_sockaddrlen = addrlen;
+  aio_connect->aio_sockaddrptr = (struct sockaddr_in*)uv__malloc(addrlen);
+  if (aio_connect->aio_sockaddrptr == NULL)
+    return -1;
```

Comment:
Leaves the socket in non-blocking mode.  Maybe do this first, then call `uv__nonblock()` and free the memory again when it fails?

Reply:
Good catch, thanks!